lib/std/os/windows.zig

```diff
@@ -183,6 +200,26 @@ pub fn WaitForSingleObject(handle: HANDLE, milliseconds: DWORD) WaitForSingleObj
     }
 }
 
+pub fn WaitForMultipleObjects(handles: []const HANDLE, waitAll: bool, milliseconds: DWORD, alertable: bool) !u32 {
+    assert(handles.len < MAXIMUM_WAIT_OBJECTS);
+    const nCount: DWORD = @intCast(u32, handles.len);
+    return switch (kernel32.WaitForMultipleObjectsEx(
+        nCount,
+        handles.ptr,
+        @boolToInt(waitAll),
+        milliseconds,
+        @boolToInt(alertable),
+    )) {
+        WAIT_OBJECT_0...WAIT_OBJECT_0+MAXIMUM_WAIT_OBJECTS => |n| if (n - WAIT_OBJECT_0 <= nCount) n else error.Unexpected,
+        WAIT_ABANDONED_0...WAIT_ABANDONED_0+MAXIMUM_WAIT_OBJECTS => |n| if (n - WAIT_ABANDONED_0 <= nCount) error.WaitAbandoned else error.Unexpected,
```

Comment:
Are we throwing away information here? I wouldn't it be useful to know which one was abandoned? Is this even an error case? Maybe the return type should be a tagged union?

Reply:
We are.... but I've never ever seen code that takes advantage of this information; nor can I think of a circumstance where it would matter. Microsoft themselves coalesce these errors into a general `AbandonedMutexException` in dotnet: https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle.waitany?view=netframework-4.8